src/containers/WrongChainRecovery/WrongChainRecovery.tsx

```diff
@@ -1,4 +1,151 @@
+import { CoinsSelectAutocomplete } from '~/components/CoinsSelectAutocomplete';
+import { useAlertBanner } from '~/contexts';
+import { mapSdkErrorToAlert, safeEnv } from '~/helpers';
+import {
+  allCoinMetas,
+  CoinMetadata,
+  wrongChainRecoveryCoins,
+} from '~/helpers/config';
+import { WrongChainRecoveryForm } from './WrongChainRecoveryForm';
+
 export function WrongChainRecovery() {
-  // TODO(BG-67209)
-  return <>{'Coming Soon'}</>;
+  const { env } = useParams<'env'>();
+  const environment = safeEnv(env);
+  const [, setAlert] = useAlertBanner();
+
+  const sourceCoins: readonly CoinMetadata[] = Object.keys(
+    wrongChainRecoveryCoins[environment]
+  ).map(key => {
+    return allCoinMetas[key as keyof typeof allCoinMetas];
+  });
+  const [sourceCoin, setSourceCoin] = useState<string>(sourceCoins[0].value);
+
+  const destinationCoins = useMemo(() => {
+    return wrongChainRecoveryCoins[environment][sourceCoin];
+  }, [environment, sourceCoin]);
+  const [destinationCoin, setDestinationCoin] = useState<string>(
+    destinationCoins[0].value
+  );
+
+  useEffect(() => {
+    const newDestinationCoins =
+      wrongChainRecoveryCoins[environment][sourceCoin];
+    if (
+      newDestinationCoins.length &&
+      !newDestinationCoins.find(item => item.value === destinationCoin)
+    ) {
+      setDestinationCoin(newDestinationCoins[0].value);
+    }
+  }, [environment, sourceCoin, destinationCoin]);
+
+  const sourceCoinHelperText = () => {
+    return (
+      <span className="tw-text-gray-700">
+        This is the coin that was sent to the wrong address.
+        <br />
+        For instance, if you sent BTC to an LTC address, the source coin would
+        be BTC.
+      </span>
+    );
+  };
+
+  const destinationCoinHelperText = () => {
+    return (
+      <span className="tw-text-gray-700">
+        This is the coin of the wallet that received the source coin.
+        <br />
+        For instance, if you sent BTC to an LTC address, the destination coin
+        would be LTC.
+      </span>
+    );
+  };
+
+  const navigate = useNavigate();
+
+  const apiProvider = (): 'Block Chair' | undefined => {
+    if (
+      sourceCoin == allCoinMetas.btc.value ||
+      sourceCoin === allCoinMetas.tbtc.value
+    ) {
+      return undefined;
+    }
+    return 'Block Chair';
+  };
+
+  return (
+    <div>
+      <div className="tw-flex tw-flex-col tw-gap-4 md:tw-flex-row md:tw-gap-16 tw-mb-4">
+        <CoinsSelectAutocomplete
+          coins={sourceCoins}
+          selectedCoin={sourceCoin}
+          onChange={event => {
+            setSourceCoin(event.currentTarget.value);
+          }}
+          helperText={sourceCoinHelperText()}
+        />
+        <CoinsSelectAutocomplete
+          coins={destinationCoins}
+          selectedCoin={destinationCoin}
+          onChange={event => {
+            setDestinationCoin(event.currentTarget.value);
+          }}
+          helperText={destinationCoinHelperText()}
```

Comment:
Nit: I think these functions are redundant, would help to see what the helper text in the context it's being used.

Reply:
Fixed